File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    # num이 1이 될 때까지 홀/짝에 맞게 작업 반복하며 작업 수 카운팅
    # 만약 cnt가 500까지 1이 되지 않는다면 -1반환
    cnt = 0
    while num != 1:
        if cnt == 500:
            return -1
        if num % 2 == 0:
            num = num/2
        else:
            num = (num*3) + 1
        cnt += 1
    return cnt

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution{
    public static void log(Object o){
        System.out.println(o);
    }
    public long collatz(long num){
        if(num % 2 == 0){
            return num/2;
        }
        else{
            return (num*3) + 1;
        }
    }
    public int solution(long num) {
        int cnt = 0;
        while(true){
            //log(num);
            // check
            if(num == 1){
                return cnt;
            }
            // before
            if(cnt == 500){
                return -1;
            }
            //collatz 
            num = collatz(num);
            cnt += 1;
        }
    }
}